File: JAVA/14_Backtracking/BOJ_2606_바이러스.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int n, m, cnt = 0;
    static boolean[][] com;
    static boolean[] visited;

    public static void dfs(int dpth) {
        visited[dpth] = true;
        for (int i = 1; i <= n; i++) {
            if (com[dpth][i] && !visited[i]) {
                cnt++;
                dfs(i);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        n = Integer.parseInt(br.readLine());
        m = Integer.parseInt(br.readLine());
        com = new boolean[n + 1][n + 1];
        visited = new boolean[n + 1];
        while (m-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            com[x][y] = true;
            com[y][x] = true;
        }
        dfs(1);
        System.out.println(cnt);
    }
}

File: C/14_Backtracking/BOJ_2606_바이러스.c
#include <stdio.h>
int n, cnt = 0, com[101][101], visited[101];
void dfs(int dpth) {
	visited[dpth] = 1;
	for (int i = 1; i <= n; i++) {
		if (com[dpth][i] && !visited[i]) {
			cnt++;
			dfs(i);
		}
	}
}
main() {
	int m, x, y;
	scanf("%d %d", &n, &m);
	while (m--) {
		scanf("%d %d", &x, &y);
		com[x][y] = 1;
		com[y][x] = 1;
	}
	dfs(1);
	printf("%d", cnt);
}